nix/haskell.nix

```diff
@@ -0,0 +1,46 @@
+############################################################################
+# Builds Haskell packages with Haskell.nix
+############################################################################
+{ lib
+, stdenv
+, pkgs
+, haskell-nix
+, buildPackages
+, config ? {}
+# GHC attribute name
+, compiler ? config.haskellNix.compiler or "ghc865"
```

Comment:
Should this really have an `or`? Shouldn't that always be defined?

Reply:
This is part of the `default` infra. I'm not actually that happy with it. The compiler shoudl come from either the stackage snapshot, the stack.yaml or the cabal.project. Overriding the compiler seems errorprone.

Reply:
I'm just saying that in the case where it's not provided I don't see why we have `or "ghc865"` rather than just failing.